### ISSUE TITLE
Scramble Matching With multiple groups in FMC produces unexpected Results

### ISSUE THREAD

**Issue by Jambrose777** [thewca/scrambles-matcher#14]
When using the scramble matcher on FMC, if the Delegate has multiple groups of scrambles; instead of producing results with multiple groups of scrambles, the result json will have 1 group with double the scrambles.

Example:
[TestingFMC.zip](https://github.com/thewca/scrambles-matcher/files/5271862/TestingFMC.zip)
I generated a Results JSON with 1 round of FMC 3 attempts (TestingFMC (1).xlsx) 
I also generated Scrambles for 1 round of FMC 2 groups (Testing FMC.json)
I entered them in the scramble matcher and automatically assigned scrambles. Here is a screenshot of that:
![image](https://user-images.githubusercontent.com/11577241/94069006-ecb2ab80-fdbd-11ea-9706-8e70e96ff6bb.png)
The results json is also included in the zip. (Results for testingFMC.json) 

On the one hand there should pretty much never be multiple scramble sets used for FMC. But on the other, this behavior of the Scramble matcher will cause errors if that were to ever happen.

If this is fixed on the scramble matcher, I think it would also be nice to include a warning in the Result Submission process so WRT can investigate when the Delegate uploads results. Issue for that: https://github.com/thewca/worldcubeassociation.org/issues/5725

**Comment by Jambrose777** [comment]
FYI: There are 22 scrambles that are of a different group in FMC. WRT will investigate these cases.

**Comment by jonatanklosko** [comment]
@viroulep I've just had a quick look and this seems to be the code responsible for that:

https://github.com/thewca/scrambles-matcher/blob/f7da683b9d7846dfd65b41a069d78d521f7d712c/src/logic/scrambles.js#L180-L197

As far as I can tell the FMC scrambles are internally split into one set per attempt, so given 2 groups of 3 attempts there are 6 internal scramble sets. The above code then merges them all into one scramble set, as if there was just one group. Any reason for handling this differently than MBLD, which handles multiple groups? (directly above the referenced code)

**Comment by jonatanklosko** [comment]
I believe this has been fixed in #15. Correct me if I'm wrong =)

**Comment by Jambrose777** [comment]
I don't believe it did. That PR fixed the issue with not being able to have multiple groups of any events due to the UI bugging out. This has to do with how FMC scrambles are generated after they have been properly selected in the UI.

**Comment by jonatanklosko** [comment]
Ah, gotcha!
